File: Java/Patient.java


import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;

public class Patient {

    private Collection<Medicine> medicines = new ArrayList<Medicine>();

    public void addMedicine(Medicine medicine) {
        this.medicines.add(medicine);
    }

    public Collection<Date> clash(Collection<String> medicineNames) {
        return clash(medicineNames, 90);
    }

    public Collection<Date> clash(Collection<String> medicineNames, int daysBack) {
        // TODO: implement this method
        return null;
    }


}

File: JavaScript/Patient.js
function Patient() {
    "use strict";

    this.medicines = [];
}

Patient.prototype.addMedicine = function (medicine) {
    "use strict";

    this.medicines.push(medicine);
};

Patient.prototype.clash = function (medicineNames, days) {
    "use strict";

    if (typeof days === "number") {
        // TODO with days
    } else {
        // TODO without days
    }
}